[New product] Alibaba ACK (#6625)
captn3m0/endoflife.date: products/alibaba-ack.md

---
title: Alibaba ACK
category: service
tags: alibaba managed-kubernetes
iconSlug: alibabacloud
permalink: /alibaba-ack
alternate_urls:
-   /ack

> -   /alibaba-container-service-for-kubernetes

Reviewer: ```suggestion
```

I don't think anybody could use or guess this URL.

versionCommand: kubectl version
releasePolicyLink: https://www.alibabacloud.com/help/en/ack/ack-managed-and-ack-dedicated/user-guide/support-for-kubernetes-versions
releaseDateColumn: true
eolColumn: End of Support

releases:
-   releaseCycle: "1.32"
    releaseDate: 2025-01-01
    eol: 2026-01-31

>     latest: '1.32.1-aliyun.1'

Reviewer: ```suggestion
```

This will be hard to maintain without automation, and automation is very hard on the CI since each release page must be scraped to get the latest version. Also we prefer also documenting `latestReleaseDate` when `latest` is set.

Is it possible to drop `latest` (and set `releaseColumn: false`) ?

-   releaseCycle: "1.31"
    releaseDate: 2024-09-01
    eol: 2025-09-30
    latest: '1.31.1-aliyun.1'

-   releaseCycle: "1.30"
    releaseDate: 2024-06-01
    eol: 2026-06-30
    latest: '1.30.7-aliyun.1'

-   releaseCycle: "1.28"
    releaseDate: 2023-10-01
    eol: 2025-10-31
    latest: '1.28.15-aliyun.1'

-   releaseCycle: "1.26"
    releaseDate: 2023-04-01
    eol: 2025-04-30
    latest: '1.26.15-aliyun.1'

-   releaseCycle: "1.24"
    releaseDate: 2022-09-01
    eol: 2024-09-30
    latest: '1.24.6-aliyun.1'

-   releaseCycle: "1.22"
    releaseDate: 2021-12-01
    eol: 2023-10-31
    latest: '1.22.15-aliyun.1'

-   releaseCycle: "1.20"
    releaseDate: 2021-04-01
    eol: 2023-04-30
    latest: '1.20.11'

-   releaseCycle: "1.18"
    releaseDate: 2020-09-01
    eol: 2022-09-30
    latest: "1.18.8"

-   releaseCycle: "1.16"
    releaseDate: 2020-02-01
    eol: 2022-06-30
    latest: "1.16.9"

---

> > [Container Service for Kubernetes (ACK)](https://www.alibabacloud.com/en/product/kubernetes) is one of the first services to pass the Certified Kubernetes Conformance Program. 
> > ACK provides high-performance management services for containerized applications. You can manage enterprise-level containerized 
> > applications throughout the application lifecycle. This service allows you to run containerized applications in the cloud in an efficient manner.

Reviewer: ```suggestion
> [Alibaba Container Service for Kubernetes (ACK)](https://www.alibabacloud.com/en/product/kubernetes) is a managed service
> that can be used to run Kubernetes on Alibaba without needing to install, operate, and maintain your
> own Kubernetes control plane or nodes. ACK runs patched Kubernetes versions and is certified Kubernetes
> conformant for a predictable experience.
```

Suggestion to make the description more neutral (inspired by https://github.com/endoflife-date/endoflife.date/blob/master/products/eks.md).

> The Kubernetes community releases a new minor version approximately every four months. Container Service for Kubernetes aligns with the upstream release cycle, iterating through the creation, maintenance, and discontinuation of Kubernetes versions. This topic describes how ACK supports various Kubernetes versions, detailing the version list update cycle, support policies, and the risks associated with using outdated Kubernetes versions.

Reviewer: From what I can see on https://www.alibabacloud.com/help/en/ack/ack-managed-and-ack-dedicated/user-guide/support-for-kubernetes-versions/, starting from 1.31 ACK provides a support cycle of one year. Did I understood this correctly ? If so could you update the description accordingly ?

> service that can be used to run Kubernetes on Alibaba Cloud without needing to install, operate, and maintain a
> Kubernetes control plane or nodes. ACK runs patched Kubernetes versions and is certified Kubernetes conformant for a
> predictable experience.

The Kubernetes community releases a new minor version approximately every four months. ACK aligns with the upstream
release cycle, iterating through the creation, maintenance, and discontinuation of Kubernetes versions.

Starting from v1.31, ACK has expanded its support for Kubernetes versions from only even-numbered minor releases,
such as v1.28 and v1.30, to include all minor versions. Additionally, for Kubernetes minor versions 1.31 and later,
the ACK support cycle has been adjusted to one year.

## Upgrading

ACK allows you to update ACK clusters from a minor version only to the following minor version. You cannot skip minor
versions when you update ACK clusters or roll back your ACK clusters to an earlier version. For example, if your cluster
runs Kubernetes 1.30 and you want to update the cluster to Kubernetes 1.32, you must first update the cluster to

> Kubernetes 1.31 and then to Kubernetes 1.32.

Reviewer: ```suggestion
ACK enables updating ACK clusters from one minor version to the next consecutive minor version.
Skipping minor versions or rolling back to an earlier version is not supported.
For example, updating a Kubernetes 1.30 cluster to Kubernetes 1.32 requires updating the cluster to Kubernetes 1.31 first.
```

Use more neutral tone.

ACK allows you to update an ACK cluster only to the latest patch version. You cannot update your cluster to outdated

> patch versions.

Reviewer: ```suggestion
ACK allows updating an ACK cluster to the latest patch version only. Updating to outdated patch versions is not possible.
```

Use more neutral tone.

## Platform Versions

The Kubernetes versions supported by ACK follow the semantic versioning scheme in the x.y.z-aliyun.n format x.y.z is the
open source Kubernetes version. x is the major version, y is the minor version, z is the patch version, and n is the ACK

> patch version.

Reviewer: ```suggestion
The Kubernetes versions supported by ACK follow the semantic versioning scheme in the `x.y.z-aliyun.n` format.
`x.y.z` is the open source Kubernetes version, and `n` is the ACK patch version.
```

Shorter, add formatting.

> After the Kubernetes community releases a new minor version, ACK performs a risk assessment and consistency test on the
> version. You can create clusters of the new version and update existing clusters to the new version within two weeks
> after the release.
> 
> After the Kubernetes community releases a new patch version for a minor version, ACK determines whether to release the
> patch version based on the risk level of the issue fixed by the patch version. If a new patch version that contains a
> high-severity vulnerability is released, ACK assesses and verifies the vulnerability within 24 hours after the release.
> After the new patch version is assessed and verified, ACK allows you to create clusters of the version and update
> clusters to the version.

Reviewer: ```suggestion
Minor Kubernetes versions are typically made available on Alibaba Cloud within two weeks of their release.

Patch Kubernetes versions may be accepted or not depending on the risk level of the fixed issues.
Patch Kubernetes versions containing high-severity vulnerability are assessed within 24 hours after their release.
```

Shorter, more neutral.
